.evergreen/config_generator/components/sasl/winssl.py

```diff
@@ -0,0 +1,100 @@
+from shrub.v3.evg_build_variant import BuildVariant
+from shrub.v3.evg_task import EvgTaskRef
+
+from config_generator.etc.function import merge_defns
+from config_generator.etc.compile import generate_compile_tasks
+
+from config_generator.etc.sasl.compile import CompileCommon
+from config_generator.etc.sasl.test import generate_test_tasks
+
+
+SSL = 'winssl'
+TAG = f'sasl-matrix-{SSL}'
+
+
+# pylint: disable=line-too-long
+# fmt: off
+COMPILE_MATRIX = [
+    ('windows-64-vs2013', 'vs2013x64', None, [       'cyrus',       ]),
+    ('windows-64-vs2013', 'vs2013x86', None, ['off',                ]),
+    ('windows-64-vs2015', 'vs2015x64', None, [       'cyrus',       ]),
+    ('windows-64-vs2015', 'vs2015x86', None, ['off',                ]),
+    ('windows-64-vs2017', 'mingw',     None, [                'sspi']),
+    ('windows-64-vs2017', 'vs2017x64', None, ['off', 'cyrus', 'sspi']),
+    ('windows-64-vs2017', 'vs2017x86', None, ['off',          'sspi']),
+]
+
+TEST_MATRIX = [
+    ('windows-64-vs2017', 'vs2017x64', None, 'off',   ['noauth', 'auth'], ['server'], [                                   'latest']),
+
+    ('windows-64-vs2013', 'vs2013x86', None, 'off',   ['noauth', 'auth'], ['server'], [       '4.0', '4.2',                       ]),
+    ('windows-64-vs2015', 'vs2015x86', None, 'off',   ['noauth', 'auth'], ['server'], [       '4.0', '4.2',                       ]),
+    ('windows-64-vs2017', 'vs2017x86', None, 'off',   ['noauth', 'auth'], ['server'], [                                   'latest']),
+
+    ('windows-64-vs2013', 'vs2013x64', None, 'cyrus', ['noauth', 'auth'], ['server'], [       '4.0', '4.2',                       ]),
+    ('windows-64-vs2015', 'vs2015x64', None, 'cyrus', ['noauth', 'auth'], ['server'], ['3.6', '4.0', '4.2',                       ]),
+    ('windows-64-vs2017', 'vs2017x64', None, 'cyrus', ['noauth', 'auth'], ['server'], [                     '4.4', '5.0', 'latest']),
+
+    ('windows-64-vs2017', 'mingw',     None, 'sspi',  [          'auth'], ['server'], [                                   'latest']),
+    ('windows-64-vs2017', 'vs2017x64', None, 'sspi',  [          'auth'], ['server'], [                                   'latest']),
+    ('windows-64-vs2017', 'vs2017x86', None, 'sspi',  [          'auth'], ['server'], [                                   'latest']),
```

Comment:
@rcsanchez97 noted that, as currently proposed, SSPI is only being tested with WinSSL. The legacy config [used to have](https://spruce.mongodb.com/version/mongo_c_driver_8b5a14e5a35fb7b365a1a53f24c933f8c497f337/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&taskName=sspi&variant=windows) compile coverage of SSPI with NoSSL and OpenSSL on some Windows distros, but they had no corresponding test tasks and were thus (currently) removed.

We could either reintroduce `sasl-sspi-(nossl|openssl|openssl-static)-compile` compile coverage, or opt to remove the `SaslSspi(NoSSL|OpenSSL|StaticOpenSSL)Compile` classes from the other SSL matrices (currently defined but not being used).

@kevinAlbs Thoughts on compile/test coverage of SSPI with SSL configs other than WinSSL?

Reply:
I expect compiling with SSPI without an SSL to be a very rare use case. SSPI is required for enterprise authentication. I expect typical users of authentication are also using SSL.

I am less sure about testing with `openssl` and `openssl-static`. But I prefer leaving them out unless there is a known need or use case motivating the additional tasks.